### PR TITLE
Fix image stats when image column type is bytes, not struct

### PR DESCRIPTION
**Pull request by polinaeterna**
should fix cases like this https://huggingface.co/datasets/lhoestq/test-cmcv

### CODE REVIEW

File: services/worker/src/worker/job_runners/split/descriptive_statistics.py [hunk 1]
```diff
@@ -583,13 +584,26 @@ class MediaColumn(Column):
...
 
+    @classmethod
+    def is_struct_type(cls, filename: Path, column_name: str) -> bool:
+        column_type = pq.read_schema(filename).field(column_name).type
+        if pa.types.is_struct(column_type):
+            return True
+        elif pa.types.is_binary(column_type) or pa.types.is_null(column_type):
+            return False
+        else:
+            raise ValueError(
```
Comment (polinaeterna) [inline services/worker/src/worker/job_runners/split/descriptive_statistics.py:602]: idk if i should do a type check and raise here actually

File: services/worker/src/worker/job_runners/split/descriptive_statistics.py [hunk 2]
```diff
@@ -656,29 +672,35 @@ class AudioColumn(MediaColumn):
...
     transform_column = IntColumn
 
     @staticmethod
-    def get_width(example: dict[str, Any]) -> int:
+    def get_width(example: Optional[Union[bytes, dict[str, Any]]], is_struct: bool = True) -> Optional[int]:
         """Get image widths."""
-        with io.BytesIO(example["bytes"]) as f:
+        if example is None:
+            return None
+        example_bytes = example["bytes"] if is_struct else example  # type: ignore
```
Comment (polinaeterna) [inline services/worker/src/worker/job_runners/split/descriptive_statistics.py:696]: `type: ignore` here because example can be both bytes and dict and mypy is not happy with bytes being indexed with string. 
any better solution to this?

File: services/worker/tests/fixtures/descriptive_statistics_dataset.py [hunk 3]
```diff
@@ -1640,9 +1640,9 @@ def nan_column() -> list[None]:
     },
     features=Features(
         {
-            "audio": Audio(sampling_rate=1600),
-            "audio_nan": Audio(sampling_rate=1600),
-            "audio_all_nan": Audio(sampling_rate=1600),
+            "audio": Audio(sampling_rate=1600, decode=False),
+            "audio_nan": Audio(sampling_rate=1600, decode=False),
+            "audio_all_nan": Audio(sampling_rate=1600, decode=False),
```
Comment (lhoestq) [inline services/worker/tests/fixtures/descriptive_statistics_dataset.py:1645]: why this change ?

File: services/worker/src/worker/job_runners/split/descriptive_statistics.py [hunk 4]
```diff
@@ -656,29 +672,35 @@ class AudioColumn(MediaColumn):
...
 
     @staticmethod
-    def get_duration(example: dict[str, Any]) -> float:
+    def get_duration(example: Optional[Union[bytes, dict[str, Any]]], is_struct: bool = True) -> Optional[float]:
         """Get audio durations"""
-        with io.BytesIO(example["bytes"]) as f:
+        if example is None:
+            return None
+        example_bytes = example["bytes"] if is_struct else example  # type: ignore
+        with io.BytesIO(example_bytes) as f:
```
Comment (lhoestq) [inline services/worker/src/worker/job_runners/split/descriptive_statistics.py:680]: I guess you can just use isinstance ?
```suggestion
        if example is None:
            return None
        elif isinstance(example, bytes):
            example_bytes = example
        else:
            example_bytes = example["bytes"]
        with io.BytesIO(example_bytes) as f:
```

File: services/worker/src/worker/job_runners/split/descriptive_statistics.py [hunk 5]
```diff
@@ -656,29 +672,35 @@ class AudioColumn(MediaColumn):
     transform_column = FloatColumn
 
     @staticmethod
-    def get_duration(example: dict[str, Any]) -> float:
+    def get_duration(example: Optional[Union[bytes, dict[str, Any]]], is_struct: bool = True) -> Optional[float]:
```
Comment (lhoestq) [inline services/worker/src/worker/job_runners/split/descriptive_statistics.py:675]: and therefore you don't need `is_struct `
```suggestion
    def get_duration(example: Optional[Union[bytes, dict[str, Any]]]) -> Optional[float]:
```

File: services/worker/src/worker/job_runners/split/descriptive_statistics.py (hunk 1 above)
Comment (lhoestq) [reply services/worker/src/worker/job_runners/split/descriptive_statistics.py:602]: if you use isinstance on the examples you might not need this function

**Review by lhoestq — COMMENTED** [review]
Cool ! Just a few comments:

File: services/worker/src/worker/job_runners/split/descriptive_statistics.py (hunk 4 above)
Comment (polinaeterna) [reply services/worker/src/worker/job_runners/split/descriptive_statistics.py:680]: yeah I first did like this but then I thought if I know in advance for all samples, why check every time. but if you think it's fine, i'd do like this

File: services/worker/tests/fixtures/descriptive_statistics_dataset.py (hunk 3 above)
Comment (polinaeterna) [reply services/worker/tests/fixtures/descriptive_statistics_dataset.py:1645]: to make `path` accessible here https://github.com/huggingface/dataset-viewer/pull/2793/files#diff-6cac2dec70bb036137022bcfb8bbe859ecc3d25b990558ef7efcb27bd75f5466R806
